libfmi: resolve broken build, update to 2.3

bstreiff opened the pull request:
Update the libfmi recipe to their 2.3 release version instead of the [last commit](https://github.com/svn2github/FMILibrary/commit/d49ed3ff2dabc6e17cc4a0c6f3fa6d2ae64a1683) from the [now-defunct svn2github](http://piotr.gabryjeluk.pl/blog:closing-svn2github) gateway, and also incorporate the [meta-ros2 layer's fix](https://github.com/ros/meta-ros/blob/master/meta-ros2/recipes-devtools/fmilibrary/fmilibrary_2.2.3.bb) for the expat-related build issues by using system expat.

bstreiff committed:
libfmi: update libfmi to 2.3

This recipe had been pointing at a "svn2github" gateway that no longer
operates. Instead, rework it based on the recipe that exists in the
meta-ros2 layer [1]. This also resolves the expat-related build issue
when this migrated to hardknott.

[1] https://github.com/ros/meta-ros/blob/master/meta-ros2/recipes-devtools/fmilibrary/fmilibrary_2.2.3.bb

Signed-off-by: Brenda Streiff <brenda.streiff@ni.com>

amstewart commented on recipes-devtools/libfmi/libfmi/0001-fmixml-use-system-expat-instead-of-building-own-with.patch:
```diff
@@ -0,0 +1,126 @@
+NOTE: VERBOSE=1 cmake --build fmilibrary/2.0.3-r0/build --target all -- -j 20
+ninja: error: build.ninja:1652: multiple rules generate ExpatEx/libexpat.a [-w dupbuild=err]
+
+Upstream-Status: Pending
```
This `.patch` was authored in June of 2020. Is it still actually `pending` upstream? Or can we bump our libfmi version and drop the .patch?

bstreiff replied:
I don't see it in their `master` or `dev_3.0` trees and it still appears to be necessary as of 2.3 (the latest release; May 2021).

amstewart commented on recipes-core/packagegroups/packagegroup-ni-extra.bb:
```diff
@@ -78,6 +78,7 @@ RDEPENDS_${PN} += "\
 	bind \
 	cifs-utils \
 	libpcap \
+	libfmi \
```
Since this package family is required by Veristand, could you move `libfmi` to `packagegroup-ni-coreimagrepo` and put it in an `RDEPENDS_${PN}_append` section with a comment declaring that it is a Veristand requirement, and tag Marcelo Izaguirre as the engineering contact.

amstewart commented on recipes-core/packagegroups/packagegroup-ni-runmode.bb:
```diff
@@ -20,6 +20,7 @@ RDEPENDS_${PN} = "\
 	glibc-gconv-cp936 \
 	glibc-gconv-iso8859-1 \
 	iproute2-tc \
+	libfmi-dev \
```
Similarly, let's break this into its own block and add the same Veristand + Marcelo comment.

bstreiff committed:
libfml: readd libfml to packagegroups

These were removed in 01b3c9ba52e3 ("libfmi: Remove libfmi packages
from packagegroups.") from the extras and runmode feeds. Since they are
needed by Veristand, lets document that fact and also move them into
the core repo.

Signed-off-by: Brenda Streiff <brenda.streiff@ni.com>
